virt_lightning/shell.py

```diff
@@ -30,26 +30,55 @@ def libvirt_callback(userdata, err):
 libvirt.registerErrorHandler(f=libvirt_callback, ctx=None)
 
 
+def _start_domain(hv, host, context, configuration):
+    if host["distro"] not in hv.distro_available():
+        logger.error("distro not available: %s", host["distro"])
+        logger.info(
+            "Please select on of the following distro: %s", hv.distro_available()
+        )
+        exit()
+
+    if "name" not in host:
+        host["name"] = re.sub(r"\W+", "", host["distro"])
```

Comment:
I don't understand why you use regexp here. Is `host["distro"]` mutable?

Reply:
"distro" is a required key. "name" is not. So, is there is no name, we create a name base on the distro.